cmake/GzInstallAllHeaders.cmake

```diff
@@ -185,31 +185,36 @@ function(gz_install_all_headers)
 
     set(component_name ${gz_install_all_headers_COMPONENT})
 
-    # Define the install directory for the component meta header
-    set(meta_header_install_dir ${GZ_INCLUDE_INSTALL_DIR_FULL}/${PROJECT_INCLUDE_DIR}/${component_name})
+    # Define the install directory for the "config" header
+    # The "meta" header will be installed one folder above this
+    set(config_header_install_dir ${GZ_INCLUDE_INSTALL_DIR_FULL}/${PROJECT_INCLUDE_DIR}/${component_name})
 
-    # Define the input/output of the configuration for the component "master" header
-    set(master_header_in ${GZ_CMAKE_DIR}/gz_auto_headers.hh.in)
-    set(master_header_out ${CMAKE_CURRENT_BINARY_DIR}/${component_name}.hh)
+    # Define the input/output of the configuration for the component "meta" header
+    set(meta_header_in ${GZ_CMAKE_DIR}/gz_auto_headers.hh.in)
+    set(meta_header_out ${CMAKE_CURRENT_BINARY_DIR}/${component_name}.hh)
 
   else()
 
-    # Define the install directory for the core master meta header
-    set(meta_header_install_dir ${GZ_INCLUDE_INSTALL_DIR_FULL}/${PROJECT_INCLUDE_DIR})
+    # Define the install directory for the "config" header
+    # The core "meta" header will be installed one folder above this
+    set(config_header_install_dir ${GZ_INCLUDE_INSTALL_DIR_FULL}/${PROJECT_INCLUDE_DIR})
 
-    # Define the input/output of the configuration for the core "master" header
-    set(master_header_in ${GZ_CMAKE_DIR}/gz_auto_headers.hh.in)
-    set(master_header_out ${CMAKE_CURRENT_BINARY_DIR}/../${GZ_DESIGNATION}.hh)
+    # Define the input/output of the configuration for the core "meta" header
+    set(meta_header_in ${GZ_CMAKE_DIR}/gz_auto_headers.hh.in)
+    set(meta_header_out ${CMAKE_CURRENT_BINARY_DIR}/../${GZ_DESIGNATION}.hh)
 
   endif()
 
-  # Generate the "master" header that includes all of the headers
-  configure_file(${master_header_in} ${master_header_out})
+  # Generate the install directory for the "meta" header one folder above the "config" header
+  cmake_path(SET meta_header_install_dir NORMALIZE ${config_header_install_dir}/..)
```

Comment:
this command was [added in cmake 3.20](https://cmake.org/cmake/help/v3.22/command/cmake_path.html), so wrap this in a `${CMAKE_VERSION}` check